bundles/org.openhab.core.crypto.cryptostore/src/main/java/org/openhab/core/crypto/cryptostore/CryptoStore.java

```diff
@@ -0,0 +1,328 @@
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link CryptoStore} is responsible for handling PKI functions.
+ *
+ * @author Ben Rosenblum - Initial contribution
+ */
+@NonNullByDefault
+public class CryptoStore {
+
+    private final Logger logger = LoggerFactory.getLogger(CryptoStore.class);
+
+    private int keySize = 128;
+    private String privKey = "";
+    private String cert = "";
+    private String caCert = "";
+    private String keystoreFileName = "";
+    private String keystoreAlgorithm = "RSA";
+    private int keyLength = 2048;
+    private String alias = "openhab";
+    private String caAlias = "trustedCa";
+    private String distName = "CN=openHAB, O=openHAB, L=None, ST=None, C=None";
+    private String cipher = "AES/GCM/NoPadding";
+    private String keyAlgorithm = "";
+
+    private @Nullable Cipher encryptionCipher;
+
+    public CryptoStore() {
+        try {
+            encryptionCipher = Cipher.getInstance(cipher);
+        } catch (NoSuchAlgorithmException | NoSuchPaddingException e) {
+            logger.debug("Could not get cipher instance", e);
+        }
+    }
+
+    public byte[] generateEncryptionKey() {
+        Key key;
+        try {
+            KeyGenerator keyGenerator = KeyGenerator.getInstance("AES");
+            keyGenerator.init(keySize);
+            key = keyGenerator.generateKey();
+            byte[] newKey = key.getEncoded();
+            this.keyAlgorithm = key.getAlgorithm();
+            return newKey;
+        } catch (NoSuchAlgorithmException e) {
+            logger.debug("Could not generate encryption keys", e);
+        }
+        return new byte[0];
+    }
+
+    private Key convertByteToKey(byte[] keyString) {
+        Key key = new SecretKeySpec(keyString, keyAlgorithm);
+        return key;
+    }
+
+    public String encrypt(String data, Key key) throws Exception {
+        return encrypt(data, key, this.cipher);
+    }
+
+    public String encrypt(String data, Key key, String cipher) throws Exception {
+        byte[] dataInBytes = data.getBytes();
+        Cipher encryptionCipher = this.encryptionCipher;
+        if (encryptionCipher != null) {
+            encryptionCipher.init(Cipher.ENCRYPT_MODE, key);
+            byte[] encryptedBytes = encryptionCipher.doFinal(dataInBytes);
+            return Base64.getEncoder().encodeToString(encryptedBytes);
+        } else {
+            return "";
+        }
+    }
+
+    public String decrypt(String encryptedData, Key key) throws Exception {
+        return decrypt(encryptedData, key, this.cipher);
+    }
+
+    public String decrypt(String encryptedData, Key key, String cipher) throws Exception {
+        byte[] dataInBytes = Base64.getDecoder().decode(encryptedData);
+        Cipher decryptionCipher = Cipher.getInstance(cipher);
+        Cipher encryptionCipher = this.encryptionCipher;
+        if (encryptionCipher != null) {
+            GCMParameterSpec spec = new GCMParameterSpec(keySize, encryptionCipher.getIV());
```

Comment:
you are using a GCMParameterSpec object here but I don't see one used in the encrypt method.  You should probably just remove it from here as the cipher you selected has NoPadding, so IV is irrelevant.  Also, invoking encryptionCipher.getIV probably just returns an empty byte array